layers/+lang/go/packages.el

```diff
@@ -93,7 +93,11 @@
 
 (when (configuration-layer/layer-usedp 'auto-completion)
   (defun go/post-init-company ()
-    (spacemacs|add-company-hook go-mode))
+    (spacemacs|add-company-hook go-mode)
+    (add-hook 'go-mode-hook
+              (lambda ()
+                (setq-local company-go-show-annotation t)
```

Comment:
`company-go-show-annotation` is part of the `company-go` package, it should be set there.